arithmetization/src/main/java/net/consensys/linea/zktracer/types/PublicInputs.java

```diff
@@ -0,0 +1,125 @@
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.zktracer.types;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static net.consensys.linea.zktracer.Trace.BLOCKHASH_MAX_HISTORY;
+import static net.consensys.linea.zktracer.Trace.LINEA_BLOB_BASE_FEE;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import lombok.experimental.Accessors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.plugin.data.BlockContext;
+import org.hyperledger.besu.plugin.data.BlockHeader;
+import org.hyperledger.besu.plugin.services.BlockchainService;
+
+@Slf4j
+@Accessors(fluent = true)
+public record PublicInputs(Map<Long, Hash> historicalBlockhashes, Map<Long, Bytes> blobBaseFees) {
+
+  public static final Bytes LINEA_BLOB_BASE_FEE_BYTES = Bytes.minimalBytes(LINEA_BLOB_BASE_FEE);
+
+  public static PublicInputs emptyPublicInputs() {
+    log.info("[ZkTracer] No public input provided, assuming line counting only or testing.");
+    return new PublicInputs(new HashMap<>(), new HashMap<>());
+  }
+
+  public static PublicInputs emptyHistoricalBlockhashes(Map<Long, Bytes> blobBaseFees) {
+    return new PublicInputs(new HashMap<>(), blobBaseFees);
+  }
+
+  public static PublicInputs defaultEmptyHistoricalBlockhashes(
+      long firstBlockNumber, long lastBlockNumber) {
+    return new PublicInputs(
+        new HashMap<>(), getDefaultBlobBaseFees(firstBlockNumber, lastBlockNumber));
+  }
+
+  public boolean allPublicInputsKnown() {
+    return historicalBlockhashes.size() >= BLOCKHASH_MAX_HISTORY && !blobBaseFees.isEmpty();
+  }
+
+  public boolean missingSomeInfos() {
+    return !allPublicInputsKnown();
+  }
+
+  // Some methods to retrieve public inputs information from BlockchainService
+  public static Map<Long, Hash> retrieveHistoricalBlockHashes(
+      BlockchainService blockchain,
+      long firstBlockNumberOfConflation,
+      long lastBlockNumberOfConflation) {
+
+    final Map<Long, Hash> historicalBlockHashes =
+        new HashMap<>(
+            (int)
+                (BLOCKHASH_MAX_HISTORY
+                    + lastBlockNumberOfConflation
+                    - firstBlockNumberOfConflation));
+
+    final long firstBlockToRetrieve =
+        Math.max(firstBlockNumberOfConflation - BLOCKHASH_MAX_HISTORY, 0);
+    final long lastBlockToRetrieve =
+        lastBlockNumberOfConflation == 0 ? 0 : lastBlockNumberOfConflation - 1;
```

Comment:
Sanity check: block number 0 corresponds to the chain's genesis block. I'm confused as to why this shouldn't be illegal / throw an exception. Do we have any tests where we "prove" the genesis block i.e. where the conflation starts at `startBlockNumber == 0`?

Hm, now that I think of it we do have certain binary flags à la `IS_GENESIS` for instance for system transactions IIRC ...